### ISSUE TITLE
Feature request: check if job names are unique in a workflow

### ISSUE THREAD

**Issue by jun-mercury** [rhysd/actionlint#578]
According to [the otel github receiver documentation](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/githubreceiver#traces---getting-started)
> IMPORTANT - Workflow Job names MUST be unique in each workflow for deterministic span IDs to not conflict with eachother. GitHub does not enforce this behavior, but when linting a workflow, warns that there are duplicate job names.

Would it be possible to add a rule to that effect?

**Comment by rhysd** [comment]
It is already reported as error.

Input:

```yaml
on: push

jobs:
  foo:
    runs-on: ubuntu-latest
    steps:
      - run: echo job1
  foo:
    runs-on: ubuntu-latest
    steps:
      - run: echo job1
```

Output:

```
test.yaml:8:3: key "foo" is duplicated in "jobs" section. previously defined at line:4,col:3. note that this key is case insensitive [syntax-check]
  |
8 |   foo:
  |   ^~~~
```
